### ISSUE TITLE
Add gazelle plugin to BCR

### ISSUE THREAD

**Issue by shs96c** [bazelbuild/bazel-skylib#410]
With #400 the gazelle plugin has now been added to the distribution, and it should be possible to ship it to BCR.

**Comment by tetromino** [comment]
After #400, I don't understand why the gazelle plugin belongs in this git repo at all:
* gazelle plugin doesn't get built by `bazel build //...`
* gazelle plugin doesn't get tested by `bazel test //...`
* gazelle plugin depends only on skylib's public api
* skylib does not depend on the gazelle plugin at all

But in any case, regardless of where the code lives, before making it releasable, we would need to add distro rules to generate the tarball. Equivalent of what we have already in distribution/BUILD for skylib.

**Comment by tetromino** [comment]
Distro rules to generate tarball will be added by #424

**Comment by tetromino** [comment]
Note: as a followup, I will try to see if we can clean up distro BUILD files to not include distro-only targets. I think this is feasible by updating to a newer version of rules_pkg, but the change is big enough that it would be best done as a separate followup PR.

**Comment by tetromino** [comment]
https://github.com/bazelbuild/bazel-central-registry/pull/403 merged 🎉
